servlet/src/main/java/io/undertow/servlet/handlers/ServletPathMatches.java

```diff
@@ -64,12 +68,49 @@
 
     private volatile ServletPathMatchesData data;
 
-    private final LRUCache<String, ServletPathMatch> pathMatchCache = new LRUCache<>(1000, -1, true); //TODO: configurable
+    private static final int CACHE_SIZE = Integer.getInteger("io.undertow.servlet.path.matches.cacheSize", 1000);
+
+    // cache for fixed data matches (no welcome files involved)
+    private final LRUCache<String, ServletPathMatch> pathMatchCacheFixed;
```

Comment:
Do you think that the performance penalty to be paid at the occasion when someone updates an exploded war is worth the penalty paid in memory, for having the two caches?

Reply:
In the default configuration I think the penalty is almost 0. Because the default configuration is -1 (cache forever) for deployed apps and 2000 (2s cache) for exploded apps (which are not recommended). So in this default conf we are good, the watch listeners are used (changes are detected in the welcome pages correctly) and everything is cached forever except if they changes.

The only problem IMO is if the customer has modified the `file-cache-time-to-live` in the `servlet-container`. In that case we are always using a TTL (no matter the type of app) and invalidating all the entries after that time. But I think we are always recommending set this to -1 and use the watch service instead.

So, I'm not sure. That's why I decided to go with this solution. But yeah, I can understand that using just one cache is quite safe. I can revert to one cache if you think it's not worthy easily.

Reply:
@rmartinc I meant memorywise, having the extra cache. Reviewing the code that uses this class, though, I see that's totally neglectable, let's keep the two caches